src/browser-realm.ts

```diff
@@ -63,55 +178,42 @@ export default function createSecureEnvironment(distortionMap?: Map<RedProxyTarg
 
     // For Chrome we evaluate the `window` object to kickstart the realm so that
     // `window` persists when the iframe is removed from the document.
-    const redWindow = (iframe.contentWindow as WindowProxy).window;
-    const { eval: redIndirectEval } = redWindow;
+    const redGlobalThis = (iframe.contentWindow as WindowProxy).window;
+    const { eval: redIndirectEval } = redGlobalThis;
     redIndirectEval('window');
+    const blueGlobalThis = globalThis;
 
     // In Chrome debugger statements will be ignored when the iframe is removed
     // from the document. Other browsers like Firefox and Safari work as expected.
     // https://bugs.chromium.org/p/chromium/issues/detail?id=1015462
     iframe.remove();
 
-    // window -> Window -> WindowProperties -> EventTarget
-    const redDocument = redWindow.document;
-    const redWindowProto = ReflectGetPrototypeOf(redWindow);
-    const redWindowPropertiesProto = ReflectGetPrototypeOf(redWindowProto);
-    const redEventTargetProto = ReflectGetPrototypeOf(redWindowPropertiesProto);
-
-    const blueDocumentProto = ReflectGetPrototypeOf(blueDocument);
-    const blueGlobalThisDescriptors = getOwnPropertyDescriptors(blueWindow) as PropertyDescriptorMap;
-    const blueWindowProtoDescriptors = getOwnPropertyDescriptors(blueWindowProto);
-    const blueWindowPropertiesProtoDescriptors = getOwnPropertyDescriptors(blueWindowPropertiesProto);
-    const blueEventTargetProtoDescriptors = getOwnPropertyDescriptors(blueEventTargetProto);
-
-    // removing problematic descriptors that should never be installed
-    delete blueGlobalThisDescriptors.location;
-    delete blueGlobalThisDescriptors.EventTarget;
-    delete blueGlobalThisDescriptors.document;
-    delete blueGlobalThisDescriptors.window;
-
-    // Some DOM APIs do brand checks for TypeArrays and others objects,
-    // in this case, if the API is not dangerous, and works in a detached
-    // iframe, we can let the sandbox to use the iframe's api directly,
-    // instead of remapping it to the blue realm.
-    // TODO [issue #67]: review this list
-    delete blueGlobalThisDescriptors.crypto;
+    const blueRefs = getCachedReferences(blueGlobalThis);
+    const redRefs = getCachedReferences(redGlobalThis);
 
     const env = new SecureEnvironment({
-        blueGlobalThis: blueWindow,
-        redGlobalThis: redWindow,
+        blueGlobalThis,
+        redGlobalThis,
         distortionMap,
     });
 
-    // other maps
-    env.remap(redDocument, blueDocument, {/* it only has location, which is ignored for now */});
-    ReflectSetPrototypeOf(redDocument, env.getRedValue(blueDocumentProto));
+    const globalDescriptors = aggregateGlobalDescriptors(
+        redRefs.windowDescriptors,
+        blueRefs.windowDescriptors,
+        endowments ? getOwnPropertyDescriptors(endowments) : {}
```

Comment:
`endowments` can be `undefined` because in `aggregateGlobalDescriptors` it's passed to `Object.assign()` as a secondary argument which is allowed to be `undefined`.